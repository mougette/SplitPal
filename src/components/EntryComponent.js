import { StatusBar } from 'expo-status-bar';
import React from 'react';
import {
    SafeAreaView,
    FlatList,
    Image,
    TouchableOpacity,
    Dimensions,
} from "react-native";
import { StyleSheet, Text, View } from 'react-native';

const Entry = (props) => {

    let color = (props.balance.startsWith("-")) ? "#bb2a2a" : "#2abb42"
    if (props.balance === "Pending") {
        color = "grey"
    }
    else if (props.balance === "Accept" || props.balance === "Split" || props.balance === "Add") {
        color = "#2a56a9"
    }

    return(
        <View style={styles.listItem}>
            <Image source ={{uri: props.image}}
                   style={{width:60, height:60,borderRadius:30}} />
            <View style={{alignItems:"center", justifyContent: 'center', flex:1}}>
                <Text style={{fontWeight:"bold"}}>{props.name}</Text>
            </View>
            <TouchableOpacity style={[styles.textButton, {backgroundColor: color}]} onPress={props.onPress}>
                <Text style={{color: "white", fontSize: 18}}>{props.balance}</Text>
            </TouchableOpacity>
        </View>
    );

}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        height: 300,
        margin: 10,
        backgroundColor: '#ffffff',
        borderRadius: 6,
        alignItems: 'center',
        justifyContent: 'center',
    },
    image: {
        height: '100%',
        borderRadius: 4,
    },
    listItem:{
        backgroundColor: "#f5f5f5",
        padding: 10,
        marginVertical: 8,
        width:400,
        flexDirection:"row",
        borderRadius: 7,
        alignItems: "center",
        justifyContent: "center",
    },
    textButton: {
        fontSize: 20,
        borderWidth: 1,
        borderRadius: 7,
        overflow: "hidden",
        padding: 10,
        marginLeft: 10,
        width: '30%',
        color: '#2196F3',
        backgroundColor: '#2abb42',
        alignItems: "center",
    },
});

export default Entry;