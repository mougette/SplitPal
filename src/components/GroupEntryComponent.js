import { StatusBar } from 'expo-status-bar';
import React, { useState, useEffect, useContext } from 'react';
import {
    SafeAreaView,
    FlatList,
    Image,
    TouchableOpacity,
    Dimensions,
} from "react-native";
import { StyleSheet, Text, View } from 'react-native';

const GroupEntry = (props) => {
const [groupStatus, setGroupStatus] = useState(true);
    const buttonPress = () =>{
    setGroupStatus(!groupStatus)
    props.onPress();
    }
    return(
        <View style={[styles.listItem, {backgroundColor: groupStatus ? "#f5f5f5": "#2a56a9"}]}>
        <TouchableOpacity onPress = {buttonPress}>
            <Image source ={{uri: props.image}}
                   style={{width:20, height:20,borderRadius:30}} />
            <View style={{alignItems:"center", justifyContent: 'center', flex:1}}>
                <Text style={{fontWeight:"bold", color: groupStatus ? "#000" : "#fff"}}>{props.name}</Text>
            </View>
         </TouchableOpacity>
        </View>
    );

}

const styles = StyleSheet.create({
    image: {
        height: '100%',
        borderRadius: 4,
    },
    listItem:{
        padding: 10,
        marginVertical: 8,
        width:100,
        flexDirection:"row",
        borderRadius: 7,
        alignItems: "center",
        justifyContent: "center",
    },
    textButton: {
        fontSize: 20,
        borderWidth: 1,
        borderRadius: 7,
        overflow: "hidden",
        padding: 10,
        marginLeft: 10,
        width: '30%',
        color: '#2196F3',
        backgroundColor: '#2abb42',
        alignItems: "center",
    },
});

export default GroupEntry;