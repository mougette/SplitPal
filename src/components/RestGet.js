export function Get(url, quearyString){
                 return fetch(url+quearyString,
                 {
                 	method: "GET"
                 })
                 .then((response) => response.json())
                 .catch((error) => console.warn("fetch error:", error))
                 .then((responseData) => {
                 if(responseData == "UserName or Password is incorrect"){
                    alert(responseData)
                 }
                   return responseData;
                 })
                 .catch(error => console.warn(error));
               }
export default {Get};