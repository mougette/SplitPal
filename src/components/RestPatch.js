 const Patch = (url, body) => {
    return fetch(url, {
      method: 'PATCH',
      body: body
    }).then((response) => response.json())
          .then((json) => {
            console.log(json)
            alert(json);
          });
  }
export default Patch;