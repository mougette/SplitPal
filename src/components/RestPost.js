export function Post(url, body) {
    return fetch(url, {
      method: 'POST',
      body: body
    })
        .then((response) => response.json())
        .catch((error) => console.warn("fetch error:", error))
        .then((responseData) => {
            return responseData;
        })
        .catch(error => console.warn(error));
  }
export default {Post};