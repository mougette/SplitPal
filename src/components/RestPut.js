export function Put(url, body) {
    return fetch(url, {
      method: 'PUT',
      body: body
    })
        .then((response) => response.json())
        .catch((error) => console.warn("fetch error:", error))
        .then((responseData) => {
            console.log(responseData)
            return responseData;
        })
        .catch(error => console.warn(error));
  }
export default {Put};
