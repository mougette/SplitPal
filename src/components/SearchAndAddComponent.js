import React from 'react';
import { TouchableOpacity, TextInput, StyleSheet, Text, View } from "react-native";

const SearchAndAdd = ( props ) => {

    return(
        <View style={styles.link}>
            <TextInput
                style={styles.input}
                placeholder={ props.search }
            />
            <TouchableOpacity onPress={() => props.onPress()}>
                <Text style={styles.blueTextButton}>{ props.button }</Text>
            </TouchableOpacity>
        </View>
    );

}

const styles = StyleSheet.create({
    link: {
        flexDirection: 'row',
        justifyContent: 'center',
        paddingTop: 20,
    },
    input: {
        width: '60%',
        borderWidth: 1,
        borderRadius: 7,
        padding: 10,
        backgroundColor: 'white',
    },
    blueTextButton: {
        fontSize: 20,
        borderWidth: 1,
        borderRadius: 7,
        overflow: "hidden",
        padding: 10,
        marginLeft: 10,
        color: '#2196F3',
        backgroundColor: 'white',
    },
});

export default SearchAndAdd;