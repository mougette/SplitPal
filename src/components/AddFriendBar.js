import React, { useState, useEffect, useContext } from 'react';
import {Post} from '../components/RestPost';
import { TouchableOpacity, TextInput, StyleSheet, Text, View } from "react-native";

const AddFriendBar = ( props ) => {
const [friend, setFriend] = useState("");
    return(
        <View style={styles.link}>
            <TextInput
                style={styles.input}
                onChangeText={setFriend}
                value={friend}
                placeholder={ props.search }
            />
            <TouchableOpacity onPress={() => Post("https://wt9b6sq6k1.execute-api.us-east-2.amazonaws.com/Iteration_1/friend-request",
                                                             JSON.stringify({
                                                              userEmail: props.email,
                                                              friendEmail: friend,
                                                              }))}>
                <Text style={styles.blueTextButton}>{ props.button }</Text>
            </TouchableOpacity>
        </View>
    );

}

const styles = StyleSheet.create({
    link: {
        flexDirection: 'row',
        justifyContent: 'center',
        paddingTop: 20,
    },
    input: {
        width: '60%',
        borderWidth: 1,
        borderRadius: 7,
        padding: 10,
        backgroundColor: 'white',
    },
    blueTextButton: {
        fontSize: 20,
        borderWidth: 1,
        borderRadius: 7,
        overflow: "hidden",
        padding: 10,
        marginLeft: 10,
        color: '#2196F3',
        backgroundColor: 'white',
    },
});

export default AddFriendBar;