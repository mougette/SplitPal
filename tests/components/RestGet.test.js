import {Get} from '../../src/components/RestGet';
import React from 'react';
import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react-native'
global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve("UserName or Password is incorrect"),
  })
  );

describe('Test Hooks', () => {
    it('positive', () => {

    expect(Get()).toMatchSnapshot();
    });
    })