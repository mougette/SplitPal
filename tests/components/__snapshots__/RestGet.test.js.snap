// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Hooks positive 1`] = `
Promise {
  "_U": 0,
  "_V": 0,
  "_W": null,
  "_X": null,
}
`;
